src/internal/operations/common.go

```diff
@@ -20,18 +21,23 @@ func getBackupAndDetailsFromID(
 	detailsStore streamstore.Reader,
 	errs *fault.Bus,
 ) (*backup.Backup, *details.Details, error) {
-	dID, bup, err := ms.GetDetailsIDFromBackupID(ctx, backupID)
+	bup, err := ms.GetBackup(ctx, backupID)
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "getting backup details ID")
 	}
 
 	var (
-		deets details.Details
-		umt   = details.UnmarshalTo(&deets)
+		deets     details.Details
+		umt       = details.UnmarshalTo(&deets)
+		detailsID = bup.DetailsID
 	)
 
-	if err := detailsStore.Read(ctx, dID, umt, errs); err != nil {
-		return nil, nil, errors.Wrap(err, "getting backup details data")
+	if len(detailsID) == 0 {
+		return bup, nil, clues.New("no details in backup").WithClues(ctx)
+	}
+
+	if err := detailsStore.Read(ctx, detailsID, umt, errs); err != nil {
+		return nil, nil, errors.Wrap(err, "reading backup details")
```

Comment:
nit: should be clues wrap?
```suggestion
		return nil, nil, clues.Wrap(err, "reading backup details")
```

Reply:
sixes, since we aren't adding structured data, it isn't necessary one way or the other.  And I still haven't made an "everything should be clues" pass on the code.